src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.cpp

```diff
@@ -0,0 +1,291 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Containers/Array.hpp>
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Types/Integer.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Size.hpp>
+#include <OpenSpaceToolkit/Core/Utilities.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Manager.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Scale.hpp>
+
+namespace NRLMSISE00_c
+{
+extern "C"
+{
+#include <NRLMSISE-00/nrlmsise-00.h>
+}
+}  // namespace NRLMSISE00_c
+
+#include <cmath>
+
+namespace ostk
+{
+namespace physics
+{
+namespace environment
+{
+namespace atmospheric
+{
+namespace earth
+{
+
+using ostk::core::types::Integer;
+using ostk::core::types::Real;
+using ostk::core::types::Size;
+using ostk::core::ctnr::Array;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::time::Scale;
+using ostk::physics::time::Instant;
+using ostk::physics::time::DateTime;
+using ostk::physics::time::Date;
+using ostk::physics::time::Time;
+
+using EarthCelestialBody = ostk::physics::env::obj::celest::Earth;
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
+using ostk::physics::environment::atmospheric::earth::Manager;
+
+NRLMSISE00::NRLMSISE00()
+    : Model()
+{
+}
+
+NRLMSISE00* NRLMSISE00::clone() const
+{
+    return new NRLMSISE00(*this);
+}
+
+bool NRLMSISE00::isDefined() const
+{
+    return true;
+}
+
+void NRLMSISE00::computeApArray(double* outputArray, const Instant& anInstant) const
```

Comment:
Since this function is not directly used by the header. Why are we using a double* output Array? Can't we just use a Array<Real>, VectorXd or a Struct?

Reply:
Good callout. The original idea was to use the "native" data structure for NRLMSISE to avoid needing to perform a bunch of copy assignments. Then some things changed when I tried to remove the NRLMSISE import in the header...
You may be right. Perhaps this doesn't make sense anymore. 🤔

Reply:
I think my original logic still makes sense. I was kinda going for 2 things:
- keep the data structures defined in the same way as the original
- declare the data structure objects at the top level, and have sub-functions fill them in.

So using an array here would either mean adding an extra copy or modifying this [line](https://github.com/open-space-collective/open-space-toolkit-physics/pull/145/files#diff-0d149128373a9c3016ed6ba61f7ac6f56835383ee82783e73ab9f141b134bbbbR95) to use an Array type. 

Not saying that's the best design choice, but my original logic at least holds.

I'm really not that satisfied with the whole "redefine the data structures then copy data out" anyway. So I'm gonna think about this one a bit more.

Reply:
Oh man silly me I forgot I can just use the heap.

Reply:
Alright, check this out. I discovered that a reinterpret_cast works here so we can avoid the copying :mr_clean: